.github/workflows/ci-cd.yml

```diff
@@ -1,31 +1,61 @@
+---
 name: CI/CD
 
 on:
   push:
     branches:
-      - 'master'
-    tags: [ '[0-9]+.[0-9]+.[0-9]+' ]
+    - 'master'
+    tags:
+    - '[0-9]+.[0-9]+.[0-9]+'
   pull_request:
     branches:
-      - 'master'
-      - '[0-9].[0-9]+.[0-9]+'
+    - 'master'
+    - '[0-9].[0-9]+.[0-9]+'
 
 env:
   FORCE_COLOR: 1
 
 jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.x"
+    - name: Install build dependencies
+      run: python -m pip install build twine
+    - name: Build distribution 📦
+      run: python -m build
+    - name: Check distribution 📦
+      run: python -m twine check --strict dist/*
+    - name: Upload distribution 📦
+      uses: actions/upload-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+
   test:
     name: Test Python ${{ matrix.python-version }}
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', '3.13']
+        python-version:
+        - '3.8'
+        - '3.9'
+        - '3.10'
+        - '3.11'
+        - '3.12'
+        - '3.13'
         include:
-          - aiohttp-version: '==3.9.2'
-          - aiohttp-version: '<4.0.0'
-            python-version: '3.11'
+        - aiohttp-version: '==3.9.2'
+        - aiohttp-version: '<4.0.0'
+          python-version: '3.11'
```

Comment:
I personally dislike quoting all the unnecessary places. It's inconsistent: we don't quote `name:` or `runs-on:` values, but do this for arbitrary strings — that's two styles in one.

Whenever I need something as an explicit string, I use a `>-` which allows me to put stuff on a separate line, drop the quotes and I don't have to escape any quotes that may be met inside such strings so it looks cleaner.